src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClient/LocalDBAPI.cs

```diff
@@ -10,25 +10,41 @@ namespace Microsoft.Data
 {
     internal static partial class LocalDBAPI
     {
-        private const string const_localDbPrefix = @"(localdb)\";
+        private const string LocalDbPrefix = @"(localdb)\";
+        private const string LocalDbPrefix_NP = @"np:\\.\pipe\LOCALDB#";
 
 
         [UnmanagedFunctionPointer(CallingConvention.Cdecl, CharSet = CharSet.Unicode)]
         private delegate int LocalDBFormatMessageDelegate(int hrLocalDB, uint dwFlags, uint dwLanguageId, StringBuilder buffer, ref uint buflen);
 
         // check if name is in format (localdb)\<InstanceName - not empty> and return instance name if it is
+        // localDB can also have a format of np:\\.\pipe\LOCALDB#<some number>\tsql\query
         internal static string GetLocalDbInstanceNameFromServerName(string serverName)
         {
-            if (serverName == null)
-                return null;
-            serverName = serverName.TrimStart(); // it can start with spaces if specified in quotes
-            if (!serverName.StartsWith(const_localDbPrefix, StringComparison.OrdinalIgnoreCase))
-                return null;
-            string instanceName = serverName.Substring(const_localDbPrefix.Length).Trim();
-            if (instanceName.Length == 0)
-                return null;
-            else
-                return instanceName;
+            if (serverName is not null)
```

Comment:
The end of the server name is never trimmed. If a server name of "(localdb)\exampleInstance       " is used, then "exampleInstance       " is returned. Will it still connect?

Reply:
I changed it to Trim. Same as managed SNI approach.

Reply:
changed it to trim